insights/v1alpha1/types_insights.go

```diff
@@ -183,10 +253,13 @@ type InsightsReport struct {
 	// healthChecks provides basic information about active Insights health checks
 	// in a cluster.
 	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=100
 	// +optional
 	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
-	// uri provides the URL link from which the report was downloaded.
-	// +kubebuilder:validation:Pattern=`^https:\/\/\S+`
+	// uri is optional field that provides the URL link from which the report was downloaded.
+	// The link must be a valid HTTPS URL and the maximum length is 2048 characters.
+	// +kubebuilder:validation:XValidation:rule=`self.matches("^https://[^\\s]+")`,message=`URI must be a valid HTTPS URL (e.g., https://example.com)`
```

Comment:
Could we use the CEL URL library here instead? It has more accurate validation, and should be able to ratchet within invalid values (test for this would be useful)

Reply:
I am not sure, because we want to allow only the HTTPS URLs. If I am not overlooking anything to do so with the URL lib I would need to use `isURL && matches("^https://[^\\s]+")`. In that case I would continue using only the regex

Reply:
You should be able to do something similar to this: https://github.com/operator-framework/operator-controller/blob/58eec48882a5e445d4d05085571b92e73a4148de/api/v1/clustercatalog_types.go#L212 (using the `.getScheme()` and comparing the value)

Reply:
Cool, thanks